Hw12

Solpostgit opened the pull request:
Проверьте ДЗ. Программа работает, но классы Author и Book отдельные.

Не понятно как добавить класс Author в класс Book. Примеров в теории не было. 
public Book(String bookName, int publishingYear) {
this.bookName = bookName;
this.publishingYear = publishingYear;
а как прописать здесь класс Author?

Solpostgit committed:
create class App

Solpostgit committed:
create and filled class Author

Solpostgit committed:
Rename class App > Book

Solpostgit committed:
filled class Book

Solpostgit committed:
filled Main and add Authors, Books

Solpostgit committed:
Input output Main

Solpostgit committed:
Add comments Не понятно как прописать класс Author в классе Book.

artem-petrosyan commented on src/Book.java:
```diff
@@ -0,0 +1,18 @@
+public class Book {
+    private String bookName;
```
Добавить автора нужно по аналогии с тем, как добавлено поле bookName.
Только bookName у тебя поле типа String.
А новое поле автора будет типа Author

Solpostgit replied:
Разобрался как добавить!
Я растерялся, потому что, воспринял добавление объекта класса Author в класс Book, как добавление метода.
А это уже был следующий урок.
Спасибо за помощь!

artem-petrosyan commented on src/Book.java:
```diff
@@ -0,0 +1,18 @@
+public class Book {
+    private String bookName;
+    private int publishingYear;
+    public Book(String bookName, int publishingYear) { //должно быть так, public Book(String bookName, int publishingYear, Author author) {
```
Соответсвенно, в конструкторе появится новый параметр типа Author

artem-petrosyan commented on src/Book.java:
```diff
@@ -0,0 +1,18 @@
+    public Book(String bookName, int publishingYear) { //должно быть так, public Book(String bookName, int publishingYear, Author author) {
+        this.bookName = bookName;
+        this.publishingYear = publishingYear;
+        // не понятно, как здесь прописать класс Author
```
И тут мы аналогично присваиваем новому полю переданное значение из параметра

Solpostgit committed:
1 Исправил ошибки в классе Author;
2 Исправил ошибки в классе Book;
3 Указал в классе Book, объект класса Author;
4 В классе Main, исправил ошибки в контрукторах Author, Book;
5 Проверил работу программы, программа работает.
